tls/s2n_client_extensions.c

```diff
@@ -229,27 +218,74 @@ static int s2n_recv_client_server_name(struct s2n_connection *conn, struct s2n_s
     return 0;
 }
 
-static int s2n_recv_client_signature_algorithms(struct s2n_connection *conn, struct s2n_stuffer *extension)
+int s2n_send_client_signature_algorithms_extension(struct s2n_connection *conn, struct s2n_stuffer *out)
```

Comment:
Having both `s2n_send_client_signature_algorithms_extension` and `s2n_send_client_signature_algorithms` is pretty confusing. I suggest we only expose `s2n_send_client_signature_algorithms` and then make the function that send the extension data into static helper.